pkg/tendermint/rpc.go

```diff
@@ -301,11 +304,51 @@ func (rpc *RPC) GetWalletBalance(
 	}), &info, nil
 }
 
+func (rpc *RPC) GetConsumerAssignedKey(
+	valcons string,
+	chainID string,
+	ctx context.Context,
+) (*types.AssignedKeyResponse, *types.QueryInfo, error) {
+	if !rpc.ChainQueries.Enabled("assigned-key") {
+		return nil, nil, nil
+	}
+
+	childQuerierCtx, span := rpc.Tracer.Start(
+		ctx,
+		"Fetching validator assigned key",
+		trace.WithAttributes(
+			attribute.String("valcons", valcons),
+			attribute.String("chain-id", chainID),
+		),
+	)
+	defer span.End()
+
+	url := fmt.Sprintf(
+		"%s/interchain_security/ccv/provider/validator_consumer_addr?chain_id=%s&provider_address=%s",
+		rpc.ChainHost,
+		chainID,
+		valcons,
+	)
+
+	var response *types.AssignedKeyResponse
+	info, err := rpc.Get(url, &response, childQuerierCtx)
+	if err != nil {
+		return nil, &info, err
+	}
+
+	if response.Code != 0 {
+		info.Success = false
+		return &types.AssignedKeyResponse{}, &info, fmt.Errorf("expected code 0, but got %d", response.Code)
+	}
+
+	return response, &info, nil
+}
```

Comment:
@coderabbitai there's no tests here yet, I am going to add them later

Reply:
@coderabbitai that probably won't work as this RPC class is just a wrapper that actually does requests, which I want to avoid in CI, as it'll make it dependant on external factors. Still really useful though, thanks for the help!